autofmt.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"bytes"
+	"context"
+
+	"oss.terrastruct.com/xdefer"
+
+	"oss.terrastruct.com/d2/d2format"
+	"oss.terrastruct.com/d2/d2parser"
+	"oss.terrastruct.com/d2/lib/xmain"
+)
+
+func autofmt(ctx context.Context, ms *xmain.State) (err error) {
+	defer xdefer.Errorf(&err, "autofmt failed")
+
+	ms.Opts = xmain.NewOpts(ms.Env, ms.Log, ms.Opts.Flags.Args()[1:])
+	if len(ms.Opts.Args) == 0 {
+		return xmain.UsageErrorf("fmt must be passed the file to be formatted")
+	} else if len(ms.Opts.Args) > 1 {
+		return xmain.UsageErrorf("fmt only accepts one argument for the file to be formatted")
```

Comment:
"accepts only one"

Reply:
i think u can get rid of the other err msg too and just combine into this. e.g. `if len(ms.Opts.Args) != 1`

Reply:
nah it's not clear if just one, will just update the grammar.